servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/ContentCodecBuilder.java

```diff
@@ -29,7 +29,7 @@
      * @return {@code this}
      * @see <a href="https://en.wikipedia.org/wiki/Zip_bomb">Zip Bomb</a>
      */
-    ContentCodecBuilder setMaxAllowedPayloadSize(int maxAllowedPayloadSize);
```

Comment:
(not necessarily to fix in the PR, just general comment)
having a builder interface in the `api` package isn't unique to this case, but I think we can simplify in the future by following this pattern:
- interface in the API package (e.g. `ContentCodec`)
- builder implementation closes to implementation as possible (e.g. if the builder impl is generic and needs to be used across multiple package `api` maybe good `*SslConfigBuilder`, but if it is tied to `netty` it can live in the corresponding `netty` package)

Having a builder interface doesn't really provide much value, builders are typically tied to an implementation and don't need to share an interface (where as the thing they build does need to share a common interface).

https://github.com/apple/servicetalk/pull/1376/files#r579151766
> this API has some mismatches with Netty's meaning.

this type of mis-match would be easier to correct the closer the builder is to the implementation.

Reply:
Agreed that if the interface exposes Netty semantics or is a veneer on Netty APIs then it should be in the Netty package. If it is in the api package then there is a temptation to over-abstract and create these kind of mismatches where there no actual intent to ever have more than the Netty implementation.

Reply:
Yup agreed. We started with the impl being in the API pkg but now this becomes more visible. I am deprecating the old API so we can keep them colocated.

Reply:
agreed, +1